### PR TITLE
Fix bug where genWeight wasn't applied

### PR DESCRIPTION
**Pull request by kdlong**
The genWeight wasn't actually applied properly, which screwed up the MC normalisation.

### CODE REVIEW

**Comment by kdlong** [comment]
Believe it's working now. I should have noticed this from the plotting. 

Bugged: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/2023_03_28/2be6ae6/ptll_nominal_uncorr_all_z.pdf
Fixed: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/2023_03_29/0a15a1a/ptll_nominal_uncorr_all_z.pdf

File: wremnants/theory_tools.py [hunk 1]
```diff
@@ -242,7 +242,7 @@ def build_weight_expr(df, exclude_weights=[]):
     valid_cols = df.GetColumnNames()
     if nominal_weight not in valid_cols:
         raise ValueError(f"The weight '{nominal_weight}' must be defined in the histmaker!")
-    weights = ["central_pdf_weight", "theory_corr_weight", "exp_weight"]
+    weights = [nominal_weight, "central_pdf_weight", "theory_corr_weight", "exp_weight"]
```
Comment (davidwalter2) [inline wremnants/theory_tools.py:245]: Could we rename this to "base_weight" to avoid confusion with the "nominal_weight" column?

Comment (davidwalter2) [reply wremnants/theory_tools.py:245]: Or just use the string "weight" two times

**Comment by kdlong** [comment]
One thing I've realised is that the commit hash used in the CI doesn't actually match the ones in the PR, so it's hard to tell which is which. I'm trying to fix this now with #175

**Comment by kdlong** [comment]
Should be ready to merge now

**Pull request merged**